proxies/nodejs/handlers/location.ts

```diff
@@ -0,0 +1,140 @@
+import Koa from 'koa'
+import { DVCClient, DVCUser } from '@devcycle/nodejs-server-sdk'
+import { getEntityModelFromType, getEntityNameFromType } from '../entityTypes'
+
+//HTTP request comes in as string
+type LocationRequestBody = {
+  command: string;
+  params: string; //[{value: string | number | boolean} | {location: string} | {callbackUrl: string}] 
+  isAsync: boolean;
+};
+
+export const handleLocation = async (
+    ctx: Koa.ParameterizedContext,
+    data: {
+    clients: { [key: string]: DVCClient };
+    users: { [key: string]: DVCUser };
+    commandResults: { [key: string]: any };
+  }
+) => {
+    const body = ctx.request.body as LocationRequestBody
+    const entity = getEntityFromLocation(ctx.request.url, data)
+
+    if (entity === undefined) {
```

Comment:
you can make these first two if statements guard statements as well, i.e. returning early so you can make the try catch on the root level of the function

Reply:
Thoughts on perhaps moving these checks to a middleware function? It could also attach the entity to the request body. It would really clean up the handler

Reply:
this middleware would probably be only specific to this endpoint, no?

Reply:
yeah, its just a convenient way to separate the logic out

Reply:
would you like me to add the middleware logic?

Reply:
added ✅